code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -359,4 +359,407 @@ Difficulty: Very Hard
+		var/mob/living/L = target
+		if(L.stat < DEAD)
+			L.heal_overall_damage(heal_power, heal_power, 1)
+			PoolOrNew(/obj/effect/overlay/temp/heal, list(get_turf(target), "#80F5FF"))
+
+/mob/living/simple_animal/hostile/lightgeist/Life()
+	if(!ckey)
+		death()
+	..()
+
+
+/obj/machinery/anomalous_crystal/refresher //Deletes and recreates a copy of the item, "refreshing" it. Only works once per item.
+	activation_method = "touch"
+	cooldown_add = 50
+
+/obj/machinery/anomalous_crystal/refresher/ActivationReaction(mob/user, method)
+	if(..())
+		var/list/L
+		var/turf/T = get_step(src, dir)
+		for(var/i in T)
+			if(istype(i, /obj/item) && !istype(i, /obj/item/weapon/storage) && !(i in affected_targets) && !istype(i, /obj/item/weapon/implant) && !istype(i, /obj/item/weapon/implanter) && !istype(i, /obj/item/weapon/disk/nuclear))
+				var/obj/item/W = i
+				if(!W.admin_spawned)
+					L += W
+		var/obj/item/CHOSEN = pick(L)
+		if(CHOSEN)
+			var/A = new CHOSEN.type(T)
+			affected_targets += A
+			qdel(CHOSEN)
+
+/obj/machinery/anomalous_crystal/possessor //Allows you to bodyjack small animals, then exit them at your leisure, but you can only do this once per activation. Because they blow up. Also, if the bodyjacked animal dies, SO DO YOU.
+	activation_method = "touch"
+
+/obj/machinery/anomalous_crystal/possessor/ActivationReaction(mob/user, method)
+	if(..())
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			var/mobcheck = 0
+			for(var/i in range(1, src))
+				if(istype(i, /mob/living/simple_animal))
+					var/mob/living/simple_animal/A = i
+					if(A.melee_damage_upper > 5 || A.mob_size >= MOB_SIZE_LARGE || A.ckey || A.stat)
+						break
+					A.verbs -= /mob/living/verb/pulled
+					var/obj/structure/closet/stasis/S = new /obj/structure/closet/stasis(A)
+					H.loc = S
+					H.disabilities += MUTE
+					H.status_flags |= GODMODE
+					H.mind.transfer_to(A)
+					var/obj/effect/proc_holder/spell/targeted/exit_possession/P = new /obj/effect/proc_holder/spell/targeted/exit_possession
+					A.mind.AddSpell(P)
+					mobcheck = 1
+					break
+			if(!mobcheck)
+				new /mob/living/simple_animal/cockroach(loc) //Just in case there aren't any animals on the station, this will leave you with a terrible option to possess if you feel like it
+
+/obj/structure/closet/stasis //Abstract game mechanic workarounds are the best
+	name = "quantum entanglement stasis warp field"
+	desc = "You can hardly comprehend this thing... which is why you can't see it."
+	icon_state = null //This shouldn't even be visible, so if it DOES show up, at least nobody will notice
+	density = 1
+	anchored = 1
+	health = 0
+	var/mob/living/simple_animal/holder_animal
+	var/turf/last_location
+
+/obj/structure/closet/stasis/process()
+	for(var/mob/living/L in src)
+		L.Stun(1)
+	if(holder_animal.stat == DEAD || !istype(loc, /mob/living/simple_animal))
+		dump_contents(1)
+		holder_animal.gib()
+
+/obj/structure/closet/stasis/New(mob/user)
```

Comment:
This is... weird, if not technically wrong; I'd prefer you explicitly have the loc argument rather than consider the loc to be the user/user to be the loc(even though that's not *wrong*, exactly)